src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,96 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int MAX_SIZE_ARRAY = 10;
+    private T[] myList;
+    private int size;
+
+    public ArrayList() {
+        myList = (T[]) new Object[MAX_SIZE_ARRAY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size + 1 >= myList.length) {
+            expandingMaxSize();
+        }
+        myList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (size + 1 == MAX_SIZE_ARRAY) {
+            expandingMaxSize();
+        }
+        System.arraycopy(myList, index, myList, index + 1, size++ - index);
+        myList[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index + 1);
+        return myList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index + 1);
+        myList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
```

Comment:
should you be able to remove element at index that equals size?

Reply:
No I can't. This value is not included in the index check condition. But I made a mistake in the verification and now, wherever I used it in the methods, everything needs to be corrected.